src/core.ts

```diff
@@ -36,16 +39,20 @@ function assertHasNameAndCreate (
   }
 }
 
-function coearseAddress (address: any) {
+// TODO: should probably replace this with ssb-ref#toMultiServerAddress or
+// just delete this and let multiserver handle invalid addresses. The 2nd option
+// sounds better, because we might already have address validation in ssb-conn
+// and so we don't need that kind of logic in secret-stack anymore.
+function coearseAddress (address: unknown) {
   if (isPlainObject(address)) {
     let protocol = 'net'
-    if (address.host.endsWith('.onion')) {
+    if (typeof address.host === 'string' && address.host.endsWith('.onion')) {
```

Comment:
This was a possible runtime crash! `address.host` could have been undefined, so `address.host.endsWith` could have crashed.